src/firefly/js/charts/ui/options/HeatmapOptions.jsx

```diff
@@ -116,13 +126,22 @@ export function fieldReducer({chartId, activeTrace}) {
     };
 }
 
-export function TableSourcesOptions({tablesource={}, activeTrace, groupKey, orientation='horizontal'}) {
+export function TableSourcesOptions({tablesource={}, activeTrace, groupKey, chartId, orientation='horizontal'}) {
     // _tables.  is prefixed the fieldKey.  it will be replaced with 'tables::val' on submitChanges.
     const tbl_id = get(tablesource, 'tbl_id');
     const colValStats = getColValStats(tbl_id);
     const xyProps = (xOrY) => ({fldPath:`_tables.data.${activeTrace}.${xOrY}`, label: `${xOrY.toUpperCase()}:`,
         name: xOrY.toUpperCase(), nullAllowed: false, colValStats, groupKey, slotProps: {control: {orientation}}});
 
+    //on the first render, update store with the color scale value reduced from chart data
+    //otherwise FieldGroupConnector is replacing it with stale state (1st option in ListBox)
+    useEffect(()=>{
```

Comment:
This is the workaround I was talking about. Setting debugger at https://github.com/Caltech-IPAC/firefly/blob/0352891e4ff4be9b48b2c1d3bccec6230237f6a7/src/firefly/js/ui/FieldGroupConnector.jsx#L194  and https://github.com/Caltech-IPAC/firefly/blob/0352891e4ff4be9b48b2c1d3bccec6230237f6a7/src/firefly/js/ui/FieldGroupConnector.jsx#L204, you'll notice that `newInitState` has correct value for colorscale field (that came by reducing chart data) but for some reason, at line 204 the value fetched from store is stale ("Default" for undefined - which is calculated initially before reducer was called) and the following logic replaces this stale value with correct value.

Reply:
I think chart still uses a reducerFunc since it was written early on. This sometimes creates issues. We are trying to get away from this approach altogether. Therefore your solution is fine. Instead of using the dispatch, use the hook.
```js
    const {setVal,getVal} = useContext(FieldGroupCtx);
```

_then_

```js
   colorscaleName && setVal(`fireflyData.${activeTrace}.colorscale`, colorscaleName);
```

Reply:
Ok thanks - I updated it!